lib/server.js

```diff
@@ -42,8 +42,10 @@ module.exports = (config, callback) => {
     // Create a server with a host and port
     const server = new Hapi.Server({
         app: {
-            datastore: config.datastore,
-            executor: config.executor
+            pipelineFactory: config.pipelineFactory,
```

Comment:
will need to update documentation of the function with this config change

Also, I'm wondering if there is a way we can make this generic

As in we pass the values in the config as:
```
config.factories = {
   pipelineFactory: instanceOfPipelineFactory
   ...
}
``` 

And then we can loop each factory and place them in the app.

Idk if that would be a good solution, but right now it seems like this `lib/server` is completely dependent on the `bin/server` to pass in all of the factories